In [7]:
! pip install apyori

  Created wheel for apyori: filename=apyori-1.1.1-cp36-none-any.whl size=5976 sha256=40068cc0720887c60d3f90fd72b102acb9beec4abedeb585d434f5cc72b595a8
  Stored in directory: C:\Users\andimu064127\AppData\Local\pip\Cache\wheels\7b\2a\35\c0c3749c1a36d4f454ea22d8396e1b854b86340d63cbbb7949
Successfully built apyori


In [8]:
#Import the library

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from apyori import apriori

## Data Preprocessing

In [14]:
#Load the dataset

df = pd.read_csv('C:\\Users\\andimu064127\\Tugas\\MA 3\\Data\\market_gform.csv')
df.tail()

Name     Item 1  Item 2  Item 3
19             xxnx       Soap     Bag  Guitar
20            bakul     Router     Bag     NaN
21  Agrawal Srikant      Watch  Racket    Soap
22              jul  Music Pad    Soap   Watch
23        Andi Isra      Mouse  Camera    Soap

In [15]:
df.drop(['Name'],axis=1, inplace=True)

**Kenapa?**

Karena penggunaan kolom nama tidak diperlukan dalam analisis algoritma apriori, sehingga dapat kita pertimbangkan untuk dibuang.

In [16]:
df.tail()

Item 1  Item 2  Item 3
19       Soap     Bag  Guitar
20     Router     Bag     NaN
21      Watch  Racket    Soap
22  Music Pad    Soap   Watch
23      Mouse  Camera    Soap

In [19]:
# See how many rows in the dataset

recs = len(df)
print(recs)

24


In [23]:
lists = []
for i in range(len(df)):
    lists.append([str(df.values[i,j]) for j in range(0,3)])

# Algoritma Apriori

Memahami parameter dalam penggunaan algoritma apriori<br>
- **`Support`**: Support is an indication of how frequently the itemset appears in the dataset.<br>
- **`Confidence`**: Confidence is an indication of how often the rule has been found to be true.<br>
- **`Lift`**: The ratio of the observed support to that expected if X and Y were independent.<br><br>

So, we can define/assumption:
- Batas support yang ditentukan adalah 5% atau 0.05 <br>
- Minimum confidence sebesar 5% atau 0.05. <br>
- Nilai lift yang diingingkan adalah 1.15<br>

In [52]:
rules = apriori(lists, min_support=0.05, min_confidence=0.05, min_lift=1.15)
assoc_results = list(rules)

In [53]:
print(len(assoc_results))

9


In [54]:
print(assoc_results[0])

RelationRecord(items=frozenset({'Bag', 'nan'}), support=0.08333333333333333, ordered_statistics=[OrderedStatistic(items_base=frozenset({'Bag'}), items_add=frozenset({'nan'}), confidence=0.6666666666666666, lift=3.1999999999999997), OrderedStatistic(items_base=frozenset({'nan'}), items_add=frozenset({'Bag'}), confidence=0.39999999999999997, lift=3.1999999999999997)])


In [55]:
results=[]
for item in assoc_results:

    pair = item[0] 
    items = [x for x in pair]
    
    barang1=str(items[0])
    barang2=str(items[1])
    support=str(item[1])[:7]
    conf=str(item[2][0][2])[:7]
    lift=str(item[2][0][3])[:7]
    
    rows=(barang1, barang2, support, conf, lift)
    results.append(rows)
    
labels=['Barang 1', 'Barang 2','Support','Confidence','Lift' ]
data_suggest=pd.DataFrame.from_records(results, columns=labels)

In [57]:
data_suggest.sort_values(by='Lift', ascending=False)

Barang 1 Barang 2  Support Confidence     Lift
0      Bag      nan  0.08333    0.66666  3.19999
4    Watch       HP  0.16666        1.0      1.6
8    Watch   Camera  0.08333        1.0      1.6
5   Racket     Soap  0.08333    0.33333  1.59999
7   Camera   Guitar  0.08333    0.39999  1.59999
1   Camera    Mouse  0.08333      0.125      1.5
2   Guitar   Racket    0.125      0.375      1.5
6   Camera   Guitar  0.08333    0.39999  1.37142
3   Guitar     Soap  0.08333       0.25      1.2

# Conclusion

Terdapat 9 baris set list barang yang dihasilkan oleh Algoritma Apriori kita yang dimana parameternya sudah ditentukan sebelumnya. Parameter yang dibangkitkan dalam model ini adalah sebagai berikut:

- Support: 0.05 atau 5% <br>
- Confidence: 0.05 atau 5% <br>
- Lift: 1.15 <br><br>

**Interpretation:**

* List yang nilai confidencenya adalah 1 berarti bahwa setiap ada transaksi dari barang 1 maka akan diikuti barang 2 dengan persentasi 100%.

* Nilai Lift 1.6 berarti bahwa HP 1.6 kali lipat akan dibeli oleh customer yang membeli Watch.

* Raket dan gitar memiliki jumlah transaksi yang cukup banyak namun dengan kenyataan nilai confidence yang kecil, kemungkinan pembeli raket akan membeli gitar juga tidak seberapa.